src/components/EditTodoTitle/EditTodoTitle.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  handleChangeTodo: <T>(
+    callback: (id: T) => Promise<void[]>,
+    valueToCall: T,
+  ) => void;
+  todo: Todo;
+  setIsEditing: (state: boolean) => void;
+  onUpdateTodo: (todo: Todo[]) => Promise<void[]>;
+  onDeleteTodo: (id: number[]) => Promise<void[]>;
+}
+
+export const EditTodoTitle: React.FC<Props> = ({
+  onDeleteTodo,
+  handleChangeTodo,
+  todo,
+  setIsEditing,
+  onUpdateTodo,
+}) => {
+  const [titleName, setTitleName] = useState(todo.title);
+
+  const handleChangeTitle = (newTitle: string) => {
+    setTitleName(newTitle);
+  };
+
+  const inputField = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    inputField.current?.focus();
+  }, [todo]);
+
+  const checkPressedKey = (key: string) => {
+    if (key === 'Escape') {
+      handleChangeTitle(todo.title);
+      setIsEditing(false);
+    }
+  };
+
+  const handleSubmit = (event?: React.FormEvent<HTMLFormElement>) => {
+    if (event) {
+      event.preventDefault();
+    }
+
+    if (titleName === todo.title) {
+      setIsEditing(false);
+
+      return;
+    }
```

Comment:
It would be best if you trimmed your new title, because ' trim ' !== 'trim' and it starts extra rerender

Reply:
Trim the new title value to ensure accurate comparisons and prevent unnecessary re-renders. This will improve performance and maintain cleaner logic throughout your code.